hub/core/storage/s3/s3_mapper.py

```diff
@@ -0,0 +1,167 @@
+import posixpath
+import boto3
+import botocore
+from typing import Optional
+from collections.abc import MutableMapping
+from botocore.exceptions import ClientError
+from hub.util.exceptions import S3GetError, S3SetError, S3DeletionError, S3ListError
+
+
+class S3Mapper(MutableMapping):
+    def __init__(
```

Comment:
add docstring that just basically says "for internal use only (by class S3Provider)"